### PR TITLE
:sparkles: Import fappy bird

### PR DESCRIPTION
**Pull request by AubaneNourry**
This pull request introduces significant updates to the Flappy Bird Reborn project, including the addition of new components, JSON configurations for game objects, and the implementation of core game mechanics. The most important changes include setting up the CMake build system, defining JSON configurations for game objects, and implementing the behavior of key game components such as the background, bird, and pipes.

### Build System Setup:
* [`CMakeLists.txt`](diffhunk://#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR1-R35): Added CMake configuration to set up the build system, including required packages and source files.

### JSON Configurations:
* [`assets/objects/json/Background.json`](diffhunk://#diff-b79c41d6bbaa22b1afd04f365fde8b2142df8d6ad92df80a31630f3fcd4748a7R1-R38): Defined the JSON configuration for the background object, including its components and properties.
* [`assets/objects/json/Bird.json`](diffhunk://#diff-ce271d0348df35191692f7a927fd6c19f8a841d9a8e9942333d0c901b5db8763R1-R42): Defined the JSON configuration for the bird object, including its components and properties.
* [`assets/objects/json/Pipe.json`](diffhunk://#diff-81a7284f54fe94cbe1ad1dc4fc66449347fc84e232c600728945bb984bd841beR1-R38): Defined the JSON configuration for the pipe object, including its components and properties.
* [`assets/objects/json/PipeSpawner.json`](diffhunk://#diff-a8ffa06f8f2a29f29db496a109692cf9211dc4eb1e5f63ab99ca703490cefdc0R1-R14): Defined the JSON configuration for the pipe spawner object.
* [`assets/objects/json/Score.json`](diffhunk://#diff-a50d41dbb78fd1b8726ae7bace3e81187ca6c18a86a198f60cf30e26027fa855R1-R47): Defined the JSON configuration for the score object, including its components and properties.

### Component Implementations:
* `assets/objects/scripts/Background.cpp` and `assets/objects/scripts/Background.hpp`: Implemented the background behavior, including scrolling and event handling for game loss. [[1]](diffhunk://#diff-0d7cea661cacde1f3790261f99419dda9b16ce15dee47eebe78f4c0ad043716aR1-R59) [[2]](diffhunk://#diff-a2fe0cbe84710a576efa61a668241ee2fad9991446ac78e18e40b7f506c785c7R1-R130)
* `assets/objects/scripts/Bird.cpp` and `assets/objects/scripts/Bird.hpp`: Implemented the bird's behavior, including movement, jumping, and collision detection. [[1]](diffhunk://#diff-28d9f882566ab70433b063ce48c58b5f42711152e0318fc59d1086183badf70cR1-R94) [[2]](diffhunk://#diff-878db868c46e6e85d29b4d21872243324a65f79b79d6ef7cb9f6efe42abea658R1-R136)
* `assets/objects/scripts/Pipes.cpp` and `assets/objects/scripts/Pipes.hpp`: Implemented the pipes' behavior, including spawning, movement, and event handling for game loss. [[1]](diffhunk://#diff-adefb4bd0eb551e53b3a394bd26b945337f87f02c8ca071eb158b4578f0dba55R1-R123) [[2]](diffhunk://#diff-c2d5e1b82084524b8c017078a2d7c4dee501b5c920043401e30fe82f8dbec7e2R1-R162)

### CODE REVIEW

File: assets/scenes/json/Scene.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+    "id": "bf69efff-4671-4981-9790-4eb56202659a",
+    "objects": [
+      "d561fa56-9f99-459f-9888-da6fdbdc2ef4",
+      "d9e329e7-b3bf-412e-86a5-f8e18f710756",
+      "9a24f7e2-edbb-4e54-a5dc-944454c8c1fd",
+      "2527ff15-35f9-44e3-a649-69f3ba25aff1",
+      "0ed660d4-a527-42ad-be4b-f568a32948da"
+    ]
+  }
+
```
Comment (6im0n) [inline assets/scenes/json/Scene.json:11]: Missing EOF

**Review by 6im0n — APPROVED** [review]
Good for me !

File: assets/objects/scripts/Score.hpp [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+     * @return Serialized JSON data.
+     * @version v0.1.0
+     * @since v0.1.0
+     * @author Aubane Nourry
+     */
+    json::IJsonObject *serializeData() override;
+
+private:
+    std::chrono::time_point<std::chrono::system_clock> startTime; ///< Start time for score tracking
+    std::chrono::time_point<std::chrono::system_clock> actualTime; ///< Current time
```
Comment (Marius-P1) [inline assets/objects/scripts/Score.hpp]: Make it Windows compatible

File: assets/objects/scripts/Pipes.hpp [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+     * @since v0.1.0
+     * @author Landry Gigant
+     */
+    json::IJsonObject *serializeData() override;
+
+private:
+    float speed = 300.0f; ///< Speed of the pipes' movement
+    float spawnRate = 2.00f; ///< Rate at which pipes spawn
+    std::chrono::time_point<std::chrono::system_clock> startTime; ///< Start time for pipes
+    std::chrono::time_point<std::chrono::system_clock> actualTime; ///< Current time
```
Comment (Marius-P1) [inline assets/objects/scripts/Pipes.hpp]: Make it Windows compatible

File: assets/objects/scripts/Background.hpp [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+     * @version v0.1.0
+     * @since v0.1.0
+     * @author Aubane Nourry
+     */
+    json::IJsonObject *serializeData() override;
+
+private:
+    float speed = 1.00f; ///< Speed of the background scrolling
+    std::chrono::time_point<std::chrono::system_clock> startTime; ///< Start time of the background
+    std::chrono::time_point<std::chrono::system_clock> actualTime; ///< Current time
```
Comment (Marius-P1) [inline assets/objects/scripts/Background.hpp]: Make it Windows compatible

**Review by Marius-P1 — APPROVED** [review]
Juste update the chrono variables

**Pull request merged**